lightningd/io_loop_with_timers.c

```diff
@@ -26,9 +26,12 @@ void *io_loop_with_timers(struct lightningd *ld)
 		/*~ Notice that timers are called here in the event loop like
 		 * anything else, so there are no weird concurrency issues. */
 		if (expired) {
-			db_begin_transaction(ld->wallet->db);
+			/* This routine is legal in early startup, too. */
```

Comment:
I would mildly suggest a variant that has `wrap_in_transaction` flag, make `io_loop_with_timers` a wrapper around that variant.  e.g.

    static void *io_loop_with_timers_core(struct lightningd *ld, bool wrap_in_db_transaction);
    void *io_loop_with_timers_without_db(struct lightningd *ld) {
        return io_loop_with_timers_core(ld, false);
    }
    void *io_loop_with_timers(struct lightingd *ld) {
        return io_loop_with_timers_core(ld, true);
    }